src/lib/layer.js

```diff
@@ -432,6 +432,7 @@ export default class Layer {
     // Call subclass lifecycle method
     this.finalizeState();
     // End lifecycle method
+    removeLayer(this.id);
```

Comment:
The seer calls can look quite confusing here, as all the functions here already deal with layers in really subtle and interconnected ways. It would be good if one could see at a glance that these calls were related to the seer integration. 
```
removeLayerFromSeer(this.id);
```
Would make it so much easier to spot what is going on.

I already started renaming some of the other functions for the same reason.